src/Syntax.php

```diff
@@ -35,38 +42,107 @@ public function getSingleShortcodeRegex()
     public function getArgumentsRegex()
         {
         $equals = $this->quote($this->getParameterValueSeparator());
-        $string = $this->quote($this->getParameterValueDelimiter());
-
-        // lookahead test for either space or end of string
-        $empty = '(?=\s|$)';
-        // equals sign and alphanumeric value
-        $simple = $equals.'\w+';
-        // equals sign and value without unescaped string delimiters enclosed in them
-        $complex = $equals.$string.'([^'.$string.'\\\\]*(?:\\\\.[^'.$string.'\\\\]*)*?)'.$string;
+        $quote = $this->quote($this->getParameterValueDelimiter());
+        $cacheKey = "arguments-$equals-$quote";
+
+        if (!isset(self::$regexCache[$cacheKey])) {
```

Comment:
Just a friendly reminder to use `static::` rather than `self::` to leverage LSB, even though we won't have anything `static`.